src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -510,6 +521,26 @@ void ShenandoahConcurrentGC::entry_evacuate() {
   op_evacuate();
 }
 
+void ShenandoahConcurrentGC::entry_promote_in_place() {
+  shenandoah_assert_generational();
+
+  ShenandoahHeap* const heap = ShenandoahHeap::heap();
+  TraceCollectorStats tcs(heap->monitoring_support()->concurrent_collection_counters());
+
+  static const char* msg = "Promote in place";
+  ShenandoahConcurrentPhase gc_phase(msg, ShenandoahPhaseTimings::promote_in_place);
+  EventMark em("%s", msg);
+
+  ShenandoahWorkerScope scope(heap->workers(),
+                              ShenandoahWorkerPolicy::calc_workers_for_conc_evac(),
+                              "promote in place");
+
+  // Ultimately, the same closure handles in-place promotions for tenured regions. No objects will be moved.
+  // The threads will still participate in the oom-evac-protocol, although no memory will be allocated for
```

Comment:
Since there is no allocation for promote-in-place promotions, it seems there would be no need for these threads to participate in oom-evac protocol.  (But maybe they still monitor the conditions that would indicate oom-evac just so we can reuse the same code.)

Reply:
Yes, I was being lazy for the sake of code reuse. This also only happens when it is an abbreviate cycle _and_ there are regions to promote. I'll see what it looks like to not enter the oom-evac-protocol for this case.